clients/src/test/java/org/apache/kafka/clients/admin/MockKafkaAdminClientEnv.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.clients.admin;
+
+import org.apache.kafka.clients.Metadata;
+import org.apache.kafka.clients.MockClient;
+import org.apache.kafka.common.Cluster;
+import org.apache.kafka.common.utils.Time;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
```

Comment:
@cmccabe here's the `MockKafkaAdminClientEnv` class that for the most part was pulled out from your `KafkaAdminClientTest` class, making it possible to create an `AdminClient` that uses mock clients. See [this unit test](https://github.com/apache/kafka/pull/2984/files#diff-1480282b21623cc8274c96dd72f1acf1R57) for an example of how it's used to test the `TopicAdmin` component that itself uses the `AdminClient`.